pkg/alertmanager/amcfg_test.go

```diff
@@ -2380,6 +2532,10 @@ func TestGenerateConfig(t *testing.T) {
 			cb := newConfigBuilder(logger, *tc.amVersion, store, tc.matcherStrategy)
 			cb.cfg = &tc.baseConfig
 
+			if tc.expectedError {
+				require.Error(t, cb.addAlertmanagerConfigs(context.Background(), tc.amConfigs))
+				return
+			}
 			require.NoError(t, cb.addAlertmanagerConfigs(context.Background(), tc.amConfigs))
```

Comment:
(nit) this would be more idiomatic.
```suggestion
			err := cb.addAlertmanagerConfigs(context.Background(), tc.amConfigs)
			if tc.expectedError {
				require.Error(t, err)
				return
			}
			require.NoError(t, err)
```

Reply:
I will do a follow up PR :)